onnxruntime/core/common/path.cc

```diff
@@ -253,6 +253,21 @@ Path& Path::Append(const Path& other) {
   return *this;
 }
 
+Path& Path::Concat(const PathString& string) {
+  components_.back() += string;
+  return *this;
+}
+
+Path& Path::ConcatIndex(const int index) {
```

Comment:
IMO, this seems kind of specialized and like it might be better outside of the Path class, i.e. path.Concat(index_string).

Reply:
I am not sure -- the idea is that the caller should not worry about the path mingling on different platform. Path.h as the path library should take care of it for all platforms...